### PR TITLE
Update to wgpu-native v0.18.1.1

### LINKED ISSUE #352: [Qt, Windows] Crash when switching from monitor to monitor with different DPI values
When I drag the window from one monitor to another, with different DPI settings, pygfx crashes:

```
❯ python .\examples\feature_demo\colormap_channels.py
QWindowsWindow::setDarkBorderToWindow: Unable to set dark window border.
Requested size 640x480 is outside of the supported range: Extent3d { width: 1600, height: 1200, depth_or_array_layers: 1 }..=Extent3d { width: 1600, height: 1200, depth_or_array_layers: 1 }
Uncaught WGPU error (Unknown):
Outdated
thread '<unnamed>' panicked at 'Unable to end render pass: RenderPassError { scope: Pass((0, 122, Vulkan)), inner: MissingAttachments }', src\command.rs:298:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### PR DESCRIPTION
**Pull request by almarklein**
closes #352

* [x] Update the header files.
* [x] Run codegen, fix parsing.
* [x] Handle fixme's.
* [x] Refactor canvas context, removing swap chain concepts in favor of wgpuSurfaceXX (check all).
* [x] use new WGPUCompositeAlphaMode?
* [x] What is FifoRelaxed present mode?
* [x] Update tests and examples
* [x] Fix that `CanvasContext`'s  don't clean up properly. 
* [x] ~Some calls can pass null pointer, check `WGPU_NULLABLE` and check for it.~  that nullable texture only applied to that particular situation, and we've got that covered.
* [x] Check #405 -> nope, still leaks
* [x] Check #352 -> yes! 
* [x] Update release notes. 
* [x] Prepare a pygfx PR: https://github.com/pygfx/pygfx/pull/617
  * [x] Adjust to API changes
  * [x] Perhaps shader changes. *Not needed*
  * [x] Can we use features like `Float32Filterable` rather than `WGPUNativeFeature_TextureAdapterSpecificFormatFeatures` -> not yet

After this is merged:
* [ ] Release wgpu-py.
* [ ] Merge pygfx-pr, and release pygfx.

### CODE REVIEW

**Comment by almarklein** [comment]
As for #352, it looks like the warnings (when dragging a window from one monitor to another on Windows+Qt) cannot be avoided without breaking window resizing on other systems. Nevertheless, the window is not in a broken state anymore where a resize is needed to get it working again. In other words, the error messages are annoying but otherwise harmless.

**Comment by almarklein** [comment]
almost done now 😅

**Comment by almarklein** [comment]
> Can we use features like `Float32Filterable` rather than `WGPUNativeFeature_TextureAdapterSpecificFormatFeatures`

The reason why I want this is that the latter is a native-only feature (not part of the webgpu spec). This means that using it also unlocks other stuff, and accidentally using that may cause downstream code to become incompatible with WebGPU.

I tried adding it to wgpu-native (https://github.com/gfx-rs/wgpu-native/pull/322), but the feature is simply not implemented in wgpu-core yet. I'm looking into adding it there, but it will take some time before that lands in a new wgpu-native release.

**Comment by almarklein** [comment]
Fixed the pypy memytest with another well-placed `gc.collect` 😉

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,27 @@ Possible sections in each release:
 * Security: in case of vulnerabilities.
 
 
+### [v0.13.0] - TBD
+
+IN this release we
```
Comment (Korijn) [inline CHANGELOG.md:22]: I would just remove this line

Comment (almarklein) [reply CHANGELOG.md:22]: Woops. I think I started a sentence there, but made it a bulletpoint.

**Review by Korijn — APPROVED** [review]

**Pull request merged**